src/lib/RelayOrderLib.sol

```diff
@@ -4,39 +4,42 @@ pragma solidity ^0.8.0;
 import {IPermit2} from "permit2/src/interfaces/IPermit2.sol";
 import {ISignatureTransfer} from "permit2/src/interfaces/ISignatureTransfer.sol";
 import {PermitHash} from "permit2/src/libraries/PermitHash.sol";
-import {RelayOrder, Input, ResolvedInput} from "../base/ReactorStructs.sol";
+import {RelayOrder, Input, FeeEscalator} from "../base/ReactorStructs.sol";
 import {ReactorErrors} from "../base/ReactorErrors.sol";
-import {InputsLib} from "./InputsLib.sol";
+import {FeeEscalatorLib} from "./FeeEscalatorLib.sol";
 
 library RelayOrderLib {
     using RelayOrderLib for RelayOrder;
-    using InputsLib for Input[];
-    using InputsLib for ResolvedInput[];
+    using FeeEscalatorLib for FeeEscalator;
 
     string internal constant PERMIT2_ORDER_TYPE = string(
-        abi.encodePacked("RelayOrder witness)", RELAY_ORDER_TYPESTRING, PermitHash._TOKEN_PERMISSIONS_TYPESTRING)
+        abi.encodePacked(
+            "RelayOrder witness)",
+            FeeEscalatorLib.FEE_ESCALATOR_TYPESTRING,
+            RELAY_ORDER_TYPESTRING,
+            PermitHash._TOKEN_PERMISSIONS_TYPESTRING
+        )
     );
 
-    /// @dev Max amounts and token addresses are signed in the token permissions of the permit information.
+    /// @dev input token addresses are signed in the token permissions of the permit information.
     bytes internal constant RELAY_ORDER_TYPESTRING = abi.encodePacked(
         "RelayOrder(",
         "address reactor,",
         "address swapper,",
-        "uint256[] startAmounts,",
+        "uint256[] amounts,",
         "address[] recipients,",
-        "uint256 decayStartTime,",
-        "uint256 decayEndTime,",
+        "FeeEscalator fee,",
```

Comment:
Can we avoid nested 712 hashing here? What is the bare minimum that we need to hash from the fee struct? My initial thought is that we need the entire struct hashed because it contains `startTime` and `endTime`

Reply:
That's true we don't have to include the flat amounts for the input here even because they're in `permissions` right?

Reply:
Ok so now we have two structs nested within RelayOrder:
- Input (address token, uint256 amount, uint256 recipient)
- Fee (address token, uint256 startTime, uint256 endTime, uint256 startAmount, uint256 endAmount)
The only things NOT covered in the `permissions` array for P2 is `recipient` field from Input and `startTime`, `endTime`, and `startAmount` in the Fee.

Should we flatten fee here into `feeStartTime`, etc.?

Reply:
So I think all we NEED to sign over is the following:

address reactor, address swapper, address inputRecipient, uint256 feeStartTime, uint256 feeEndTime, address feeRecipient, uint256 feeStartAmount

bc for the input and the fee we already sign over the token and max amount. 

I would say if flattening is cheaper let's flatten!

Reply:
sounds good makes sense to me. I flattened it in latest commit!